### PR TITLE
Issue #154 : custom JsonView annotations support

### PR DESCRIPTION
**Pull request by sdeleuze**

### CODE REVIEW

**Comment by bclozel** [comment]
@bmeurant is preparing a PR regarding `@Transactional` issues.
Actually, the default strategy is `required=true` and **readOnly=false**; we were kind of used to `readOnly=true` because of `GenericService` in RESThub 1.x.

Could you delete some parts of your PR?
I'll point the exact lines to make this easier.

File: resthub-common/pom.xml [hunk 1]
```diff
@@ -39,6 +39,10 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-context-support</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-tx</artifactId>
```
Comment (bclozel) [inline resthub-common/pom.xml]: Could you delete this?
PR regarding `@Transactional` will handle this.

File: resthub-common/src/main/java/org/resthub/common/service/CrudServiceImpl.java [hunk 2]
```diff
@@ -16,6 +18,7 @@
  * @param <ID> Resource id type, usually Long or String
  * @param <R> The repository class
  */
+@Transactional
```
Comment (bclozel) [inline resthub-common/src/main/java/org/resthub/common/service/CrudServiceImpl.java]: Could you delete this? And all the other `@Transactional` annotations in this class.

PR regarding `@Transactional` will handle this.

**Comment by sdeleuze** [comment]
Additional informations, this functionality should allow to fix serialization issues for children-parent when we were using previously @JsonIgnore or @JsonBackReference. Another use case is that it allows us to use again lazy loading + OpenSessionInView filter.

Here is a code sample showing these 2 improvements (not tested yet, so there may be issues) :

``` java

public class Library {
    @JsonView(BookListView.class)
     private List<Book> books;
}

public class Book {

    @ManyToOne(fetch = FetchType.LAZY)
     private Library library;

    @JsonView(BookListView.class)
     private String name;

    @OneToMany(fetch = FetchType.LAZY)
    private List<Page> pages;
}

@RequestMapping("books")
@ResponseView(BookListView.class)
public @ResponseBody List<Book> getBooks()
{
    return data;
}

@RequestMapping("{id}")
public @ResponseBody Book getBook(@PathVariable("id") Integer id)
{
    return data.get(id - 1);
}

```

**Comment by bclozel** [comment]
great!

**Pull request merged**